source/extensions/filters/http/gzip/config.cc

```diff
@@ -12,27 +12,14 @@ namespace Extensions {
 namespace HttpFilters {
 namespace Gzip {
 
-Server::Configuration::HttpFilterFactoryCb
-GzipFilterFactory::createFilter(const envoy::config::filter::http::gzip::v2::Gzip& proto_config) {
-  GzipFilterConfigSharedPtr config = std::make_shared<GzipFilterConfig>(proto_config);
-  return [config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
-    callbacks.addStreamFilter(std::make_shared<GzipFilter>(config));
-  };
-}
-
-Server::Configuration::HttpFilterFactoryCb
-GzipFilterFactory::createFilterFactory(const Json::Object&, const std::string&,
-                                       Server::Configuration::FactoryContext&) {
-  NOT_IMPLEMENTED;
-}
-
 Server::Configuration::HttpFilterFactoryCb
 GzipFilterFactory::createFilterFactoryFromProto(const Protobuf::Message& proto_config,
                                                 const std::string&,
                                                 Server::Configuration::FactoryContext&) {
-  return createFilter(
-      MessageUtil::downcastAndValidate<const envoy::config::filter::http::gzip::v2::Gzip&>(
-          proto_config));
+  GzipFilterConfigSharedPtr config = std::make_shared<GzipFilterConfig>(validate(proto_config));
```

Comment:
this is a huge improvement and I think we should just move forward with this.  I also wonder here what in this file is specific to gzip, and couldn't be supplied by the framework as a template default of some sort.  But let's go with this and we can keep iterating after merge if we think there's more ways to factor out the boilerplate.

Reply:
My preference is to not partially do this, at least within a filter family. I would rather just do all of the HTTP filters so we don't leave things an an inconsistent state. Can we brainstorm additional ways to reduce boilerplate now?

The only thing that comes to mind is to have the base class which implements the functions, but then accepts lambdas in the constructors. It's possible this would git rid of some more code? Not sure how much of an improvement that would be though.